### ISSUE TITLE
display-plugin-updates-005 and -006 fail with maven older than 3.8.4

### ISSUE THREAD

**Issue by ajarmoniuk** [mojohaus/versions#850]
While working on #848 I've noticed that the project is very sensitive to concurrency issues. Integration tests often fail when run multi-threaded. To be checked and sanitized.

By the way, this is a very old issue, a comment regarding that has been sitting in pom.xml for quite some time now.

**Comment by ajarmoniuk** [comment]
artifactId can be null here:

https://github.com/mojohaus/versions/blob/master/versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java#L344

if multithreaded execution is enabled

**Comment by ajarmoniuk** [comment]
effectiveVersion is null at https://github.com/mojohaus/versions/blob/master/versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java#L407 when running with parallel threads

**Comment by slawekjaranowski** [comment]
Some clarification of how invoker works. 

For each project / test invoker execute new `process` by command line.

Sow when we use `parallelThreads > 1` we have multiple independent system process at the same time it is not `multithreaded` execution.

Each versions-m-p test can interact with local Maven repository which is not synchronized by multi process.

**Comment by ajarmoniuk** [comment]
In that case this is something I observe on my machine only; and I see it's not limited to parallel use of the invoker.

In many runs, it-display-plugin-updates-005, it-display-plugin-updates-006, it-rules-via-wagon-001 (which uses the display plugin updates mojo), all fail on an NPE at https://github.com/mojohaus/versions/blob/master/versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java#L623

because one of the versions is null, which boils down to the plugin version being null and the parentVersions not containing the version for the plugin. Looking for the root cause of that.

**Comment by slawekjaranowski** [comment]
We can use `invoker.ordinal` to try sort, group, isolate tests

https://maven.apache.org/plugins/maven-invoker-plugin/examples/parallel-projects-execution.html

**Comment by ajarmoniuk** [comment]
Managed to produce the problem **consistently** using Maven 3.2.5.
[bad.log](https://github.com/mojohaus/versions/files/10253447/bad.log)

**Comment by ajarmoniuk** [comment]
In version 3.8.6, the version of `dummy-maven-plugin` is resolved to 3.1.

```
[DEBUG] Resolved plugin version for localhost:dummy-maven-plugin to 3.1 from repository mrm-maven-plugin (http://localhost:50095, default, releases+snapshots)
```

In the older version however, it doesn't get resolved. I don't know how much consistent that is. I haven't checked how that resolution is done exactly (DefaultPluginVersionResolver from maven-core), but that might depend on the version of Maven.

**Comment by ajarmoniuk** [comment]
I think it is quite safe to mitigate (I specifically don't use the term "fix") the problem by using String.valueOf instead of simply dereferencing the String object which may be null.
